### PR TITLE
Jlc/keep subsection vars email context

### PR DESCRIPTION
**Pull request by johanv26**
## Description

The idea of this PR is used only the course_email_context to build the msgs for notications.
Is ancestor has the following fields,
https://github.com/openedx/edx-platform/blob/1e8acf49fd906d436378e70de1401f1913358921/lms/djangoapps/bulk_email/tasks.py#L111-L138
And this is a dict and would have the following extra fields:

https://github.com/eduNEXT/eox-nelp/blob/5d17d0a04396bd365adbcc290be70f2b55927a1e/eox_nelp/notifications/notify_course_due_date.py#L74-L77
``` python
{
   "course_title":"test2",
   "course_root":"/courses/course-v1:edX+cd202+2023/",
   "course_language":"en",
   "course_url":"http://lms.mango.edunext.link:8000/courses/course-v1:edX+cd202+2023/",
   "course_image_url":"http://lms.mango.edunext.link:8000/asset-v1:edX+cd202+2023+type@asset+block@images_course_image.jpg",
   "course_end_date":"",
   "account_settings_url":"http://lms.mango.edunext.link:8000/account/settings",
   "email_settings_url":"http://lms.mango.edunext.link:8000/dashboard",
   "logo_url":"http://localhost:18000/theming/asset/images/logo.png",
   "platform_name":"Nelp Mango",
   "year":2023,
   "subsection_title": "subsection-name",
   "subsection_delta": "12 days, 22:40:30.903793",
   "subsection_due_date": "2023-05-12 00:00:00+00:00"   
}
```

### Bulk Email Template

This is a template for bulk emails related to the course "{course_title}".
Course Details

    Title: {course_title}
    Language: {course_language}
    Course URL: {course_url}
    Course Image URL: {course_image_url}
    Course End Date: {course_end_date}
    Course Root: {course_root}

Subsection Details

    Title: {subsection_title}
    Due Date: {subsection_due_date}
    Time Remaining: {subsection_delta}

Platform Details

    Platform Name: {platform_name}
    Year: {year}
    Account Settings URL: {account_settings_url}
    Email Settings URL: {email_settings_url}
    Logo URL: {logo_url}

Please note that this is a general template and you may need to customize it based on your specific needs. Additionally, the values in curly braces ({}) correspond to the keys in the provided JSON data, so make sure to replace them with the correct values when using the template.

So the code should render text_plain and HTML .
Keep in mind that in admin you have to add `{{message_body}}` but for this implementation is empty "".
Also the name of the template this implementation use is `base-notification-template`.
https://edunext.stage.nelp.gov.sa/admin/bulk_email/courseemailtemplate/4/change/

eg 
#### HTML section:
``` html
<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8">
	<meta name="viewport" content="width=device-width">
	<title>{course_title} - Subsection Update</title>
</head>
<body>
	<div style="max-width: 600px; margin: auto;">
		<div style="text-align: center; padding-top: 20px;">
			<img src="{logo_url}" alt="{platform_name} Logo">
			<h2 style="margin: 0;">{course_title}</h2>
            <img style="max-block-size: 100px;" src="{course_image_url}" alt="{course_title} img">
		</div>
		<hr>
		<h3 style="margin: 0;">New Subsection Available: {subsection_title}</h3>
		<p>A new subsection is coming  to {course_title}. The new subsection, "{subsection_title}", is available now for you to explore. This subsection has a due date of {subsection_due_date}. You have {subsection_delta} to complete it before the deadline.</p>
		<p>You can access the course here: <a href="{course_url}">{course_title}</a></p>
		<p>If you have any questions or concerns, please feel free to contact us at <a href="{email_settings_url}">email_settings_url</a>.</p>
		<p>Thank you for learning with {platform_name}!</p>
		<hr>
		<p style="font-size: 12px; color: gray;">This email was sent by {platform_name}. To update your email preferences, please visit <a href="{account_settings_url}">your account settings</a>.</p>
		<p style="font-size: 12px; color: gray;">&copy; {year} {platform_name}. All rights reserved.</p>
{{message_body}}
	</div>
</body>
</html>
```
#### text plain section:
```
Notification for due dates in  {course_title}.
The course {course_title} has a notification. 
The subsection {subsection_title} is coming soon. 
The due date of this subsection is UTC {subsection_due_date}. 
There are approx {subsection_delta} for the due date of the subsection.
{{message_body}}
```
## extra
This PR adds an extra setting that can be used to render a different subject for all the platform.
https://github.com/eduNEXT/eox-nelp/blob/5d17d0a04396bd365adbcc290be70f2b55927a1e/eox_nelp/notifications/notify_course_due_date.py#L85

## Checklist for Merge

- [ ] Tested in a remote environment
- [ ] Updated documentation
- [ ] Rebased master/main
- [ ] Squashed commits

### CODE REVIEW

File: eox_nelp/notifications/tests/test_notify_course_due_date.py [hunk 1]
```diff
@@ -100,39 +91,113 @@ def test_notify_upcoming_course_due_date(self):
...
-            expected_text_msg,
-            expected_html_msg,
-            recipients_list,
+        course_title = "course-display-name"
+        subsection_title = "subsection-display-name"
+        recipients_list = ["vader@star.com", "palpatine@star.com"]
+        subsection_delta = subsection_due_date - timezone.now()
+        expected_text_msg = "expected-text-msg"
+        expected_html_msg = "expected-html-msg"
+        setattr(
```
Comment (andrey-canon) [inline eox_nelp/notifications/tests/test_notify_course_due_date.py:153]: I think it's better to use the decorator in order to revert this change after the test

Comment (johanv26) [reply eox_nelp/notifications/tests/test_notify_course_due_date.py:153]: but I am usinghttps://github.com/eduNEXT/eox-nelp/blob/0cbd232bf88f915eda14ed9ec76c4c5e1f823b70/eox_nelp/notifications/tests/test_notify_course_due_date.py#L126 @override settings decorator? is for this?

Comment (andrey-canon) [reply eox_nelp/notifications/tests/test_notify_course_due_date.py:153]: I didn't realize about it, since normally the decorator includes the new value

**Review by andrey-canon — APPROVED** [review]
